src/react-imgix.js

```diff
@@ -155,175 +145,176 @@ class ReactImgix extends Component {
+          _inPicture: true
+        })
+      ) || [];
+
+    // look for an <img> or <ReactImgix type='img'> - at the bare minimum we
+    // have to have a single <img> element or else it will not work.
+    let imgIdx = _children.findIndex(
+      c =>
+        c.type === "img" ||
+        c.type === ReactImgix ||
+        c.type === ReactImgixWrapped
     );
-    delete propsWithOutDeprecated.customParams;
 
-    if (props.faces) {
-      warning(
-        false,
-        `The prop 'faces' has been deprecated and will be removed in v9. Please update the usage to <Imgix imgixParams={{crop: 'faces'}} />`
+    if (imgIdx === -1) {
+      console.warn(
+        "No fallback <img /> or <Imgix /> found in the children of a <picture> component. A fallback image should be passed to ensure the image renders correctly at all dimensions."
       );
-      delete propsWithOutDeprecated.faces;
-      if (!imgixParams.crop) {
-        imgixParams.crop = "faces";
-      }
-    }
-    if (props.entropy) {
-      warning(
-        false,
-        `The prop 'entropy' has been deprecated and will be removed in v9. Please update the usage to <Imgix imgixParams={{crop: 'entropy'}} />`
-      );
-      delete propsWithOutDeprecated.entropy;
-      if (!imgixParams.crop) {
-        imgixParams.crop = "entropy";
-      }
+    } else if (imgIdx !== _children.length - 1) {
+      // found one, need to move it to the end
+      _children.push(_children.splice(imgIdx, 1)[0]);
     }
 
-    return <WrappedComponent {...propsWithOutDeprecated} />;
+    return <picture children={_children} />;
+  }
+}
+PictureImpl.displayName = "ReactImgixPicture";
+
+/**
+ * React component used to render <source> elements with Imgix
+ */
+class SourceImpl extends Component {
+  static propTypes = {
+    ...SHARED_IMGIX_AND_SOURCE_PROP_TYPES
+    // TODO: add media?
   };
-  WithDeprecatedProps.propTypes = {
-    ...ReactImgix.propTypes,
-    auto: PropTypes.array,
-    customParams: PropTypes.object,
-    crop: PropTypes.string,
-    entropy: PropTypes.bool,
-    faces: PropTypes.bool,
-    fit: PropTypes.string
+  static defaultProps = {
+    disableSrcSet: false,
+    onMounted: noop
   };
-  WithDeprecatedProps.defaultProps = {
-    imgixParams: {}
+
+  componentDidMount = () => {
+    const node = ReactDOM.findDOMNode(this);
+    this.props.onMounted(node);
   };
-  WithDeprecatedProps.displayName = "ReactImgix";
+  render() {
+    const { disableSrcSet, width, height } = this.props;
 
-  return WithDeprecatedProps;
-};
+    const htmlAttributes = this.props.htmlAttributes || {};
+
+    const { src, srcSet } = buildSrc({
+      ...this.props,
+      type: "source",
+      imgixParams: imgixParams(this.props)
+    });
+
+    let childProps = {
+      ...this.props.htmlAttributes,
+      sizes: this.props.sizes,
+      className: this.props.className,
+      width: width <= 1 ? null : width,
+      height: height <= 1 ? null : height
+    };
+    delete childProps.alt;
```

Comment:
What if instead of stripping the `alt` attribute, we just issued a warning that it should be included on the fallback img instead? As far as I can tell, it's not strictly illegal to include it on a `source` element, just ineffective and inappropriate.

Reply:
I'm in favour of just removing this (because it's not "strictly illegal"), and also not adding a warning. I don't think we can be expected to warn developers about everything.

Reply:
Works for me